### ISSUE TITLE
Keyboard navigation to scrolled row in Table story does not work well

### ISSUE THREAD

**Issue by elephantcatdog** [iTwin/iTwinUI#1020]
#### Describe the bug (actual behavior)
On the table story "Scroll to Row", tabbing into the table focuses on row "Name12335", a row that was out of the viewbox when the table was automatically scrolled to "Name12345 - Scrolled to me!". If you click out of the table and then tab back into it, the focus goes further back to "Name12320".

https://user-images.githubusercontent.com/37936169/213755693-1ad1fea5-ef22-4ef0-a226-0ce182ad672a.mp4

#### Expected behavior
Tabbing into the table should focus on the scrolled to row "Name12345 - Scrolled to me!".

#### Reproduction
This occurs in storybook on the table story "Scroll to Row": https://itwin.github.io/iTwinUI/react/?path=/story/core-table--scroll-to-row
*when #1018 is merged

##### Steps to reproduce
1. Click on the gray background behind the table in the story "Scroll to Row".
2. Press 'tab' five times until the focus is on an anchor "Click me!".
3. The anchor is not "Name12345 - Scrolled to me!" but "Name12335".

#### Additional information
This might just be a limitation of the table, but hopefully it could get closer.

**Comment by mayank99** [comment]
Scroll position and focus are not connected in general, so I don't think this is an issue in our Table. When _not_ using virtualization, pressing <kbd>Tab</kbd> would send you the very first row. When using virtualization, there are only a few (~10) rows in the DOM that are not currently visible, so it sends you to some seemingly "random" row.

If moving focus to the scrolled row is desired, then it can be done by app devs manually. I've seen this handy following snippet floating around which devs can call in their own code.
```js
function getFocusStartPoint() {
  return window.getSelection()?.focusNode?.parentElement;
}

function setFocusStartPoint(element: HTMLElement) {
  if (!element) return;

  const tabindex = element.getAttribute("tabindex");

  element.setAttribute("tabindex", "-1");
  element.focus();
  element.blur();

  if (tabindex) {
    element.setAttribute("tabindex", tabindex);
  } else {
    element.removeAttribute("tabindex");
  }
}
```

If there is demand for this feature, we can incorporate it into our Table or show an example. Until then, I will close this.
